src/sqlfluff/rules/L063.py

```diff
@@ -62,6 +62,7 @@ class Rule_L063(Rule_L010):
     _exclude_elements: List[Tuple[str, str]] = [
         ("type", "identifier"),
         ("type", "literal"),
+        ("type", "parameter"),
     ]
```

Comment:
I'm a little worried that there will be more edge cases like this, but unfortunately there's no easy way to distinguish all the data type keywords as not all are marked as datatype and the parent selector is picking up more than it should :-(

Reply:
Think it should be (could be) cleaned up?

Reply:
I had hoped after @alanmcruickshank fixed up TSQL data types recently, but still a lot of other ones that are just data-type->keywords or similar (e.g. the `WITH TIME ZONE` ones). So not holding out massive hope to be honest.

We could perhaps restrict the types that `parenttype` looks at, rather than allowing everything. Tried that on this branch but ended up chasing my own tail. Let me have a play with that some more before merging this.

Reply:
Actually I found a better way. We only need `parent_type` for keywords. So by restricting to that, can avoid having to maintain an exclude list.